dataset.py

```diff
@@ -22,27 +22,56 @@ def __init__(self):
         self._submit_rows = pd.read_csv(submission_rows_fn, index_col=0)
 
     
-    def preprocess(self, pp_fn='preprocess_simple', pp_fn_kwargs={}):
-        """Preprocess data using function pp_fn (with additional kwargs if necessary) from preprocessing.py"""
-        """pp_fn is string with name of preprocessing model"""
+    def preprocess(self, pp_fn=preprocess_simple, **pp_fn_kwargs):
+        """
+        Preprocess data using function pp_fn (with additional kwargs if necessary) from preprocessing.py
         
-        return eval('preprocessing.'+pp_fn+'(self._train, self._submit_rows.index, **pp_fn_kwargs)')
+        Args:
+            pp_fn: Name of preprocessing function from preprocessing.py
+            pp_fn_kwargs: Keyword arguments for preprocessing function
+            
+        Returns:
+            Output of preprocessing function applied to training data restricted to rows of interest
+        """
+        
+        return pp_fn(self._train, self._submit_rows.index, **pp_fn_kwargs)
     
     
-    def predictions(self, model_name='status_quo_model', model_kwargs={}, pp_fn='preprocess_simple', pp_fn_kwargs={}):
-        """Return predictions from model_name when fit to data preprocessed by pp_fn"""
-        """pp_fn is string with name of preprocessing model, and model_name is name of prediction model"""
-        """Allows input of additional keyword parameters for model and preprocessing functions"""
+    def predictions(self, preprocessed_data, model_name=status_quo_model,**model_kwargs):
+        """Return predictions from model_name given preprocessed data
+        
+        Args:
+            model_name: Name of prediction model from models.py
+            model_kwargs: Model function keyword arguments
+            preprocessed_data: Data formatted to be passed in for predictions
+            
+        Good idea to have option for returning pickled representation of model when defining the functions in models.py. 
```

Comment:
This could be a separate function for models that have a separate training/inference phase. We'll leave as a TODO for now.